File: src/ProjectR.Scripting/content/scripts/spells/DarkMend.lua
function GetTargetType()
  return Myself
end

function GetName()
  return "Dark Mend"
end

function GetDescription()
  return "Dark energy fills the air.\nDeals damage to enemies.\nHeals allies by 10%% of their max HP."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 1.44
end

function GetDelay()
  return .1
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { DRK }
end

function SpellEffect()
  ForEachDefenderParty("DamageFunc")
  ForEachAttackerParty("HealFunc")
end

function HealFunc(character)
  character:Heal(GetStats(character):GetTotalStat(HP) * .1)
end

function DamageFunc(character)
  local defStats = GetStats(character)
  local defMR = defStats:GetTotalStat(MR)
  local defDRK = defStats:GetTotalStat(DRK)

  local damage = (4.5 * aMD * (aDRK/100) - 1.5 * defMR) * (100/defDRK)
  character:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/DarkMend.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class DarkMend : SpellScriptBase
    {
        public override string Name { get { return "Dark Mend"; } }
        public override string Description { get { return "Dark energy fills the air.\nDeals damage to enemies.\nHeals allies by 10% of their max HP."; } }

        public override TargetType TargetType { get { return TargetType.Everyone; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.DRK, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 144; } }
        public override double Delay { get { return .1; } }

        public override void SpellEffect(ICharacter caster, IList<ICharacter> allies, IList<ICharacter> enemies)
        {
            foreach (var enemy in enemies)
            {
                Target = enemy;
                var damage = (4.5 * cMD * (cDRK / 100) - 1.5 * tMR) * (100 / tDRK);
                DealDamage(damage);
            }

            foreach (var ally in allies)
            {
                Heal(ally, HP(ally) * .1);
            }
        }
    }
}